File: list_models.js
/**
 * Gemini Model Lister
 * Fetches all available models for your API key from the v1beta endpoint.
 */

const API_KEY = process.env.GEMINI_API_KEY;

if (!API_KEY) {
  console.error("\x1b[31mError: GEMINI_API_KEY is not set.\x1b[0m");
  process.exit(1);
}

async function listAvailableModels() {
  const url = `https://generativelanguage.googleapis.com/v1beta/models?key=${API_KEY}`;

  console.log("\x1b[33mFetching available Gemini models...\x1b[0m\n");

  try {
    const response = await fetch(url);
    const data = await response.json();

    if (data.error) {
      console.error("\x1b[31mAPI Error:\x1b[0m", data.error.message);
      return;
    }

    // Filter for models that actually support content generation
    const genModels = data.models.filter(m => 
      m.supportedGenerationMethods.includes("generateContent")
    );

    console.log("ID".padEnd(35) + "DISPLAY NAME");
    console.log("-".repeat(60));

    genModels.forEach(model => {
      // Clean up the name (remove 'models/' prefix)
      const shortId = model.name.replace("models/", "");
      console.log(`\x1b[32m${shortId.padEnd(35)}\x1b[0m ${model.displayName}`);
    });

    console.log("\n\x1b[34mUsage:\x1b[0m Update MODEL_NAME in your script with one of the IDs above.");

  } catch (error) {
    console.error("\x1b[31mFetch Error:\x1b[0m", error.message);
  }
}

listAvailableModels();

File: list_models.go
package main

import (
	"context"
	"fmt"
	"os"
	"strings"

	"github.com/google/generative-ai-go/genai"
	"google.golang.org/api/iterator"
	"google.golang.org/api/option"
)

func main() {
	ctx := context.Background()
	apiKey := os.Getenv("GEMINI_API_KEY")

	if apiKey == "" {
		fmt.Println("\033[31mError: GEMINI_API_KEY is not set.\033[0m")
		os.Exit(1)
	}

	client, err := genai.NewClient(ctx, option.WithAPIKey(apiKey))
	if err != nil {
		fmt.Printf("\033[31mError creating client:\033[0m %v\n", err)
		return
	}
	defer client.Close()

	fmt.Println("\033[33mFetching available Gemini models...\033[0m\n")

	// Print Header
	fmt.Printf("%-35s %s\n", "ID", "DISPLAY NAME")
	fmt.Println(strings.Repeat("-", 70))

	// Iterate through available models
	iter := client.ListModels(ctx)
	for {
		model, err := iter.Next()
		if err == iterator.Done {
			break
		}
		if err != nil {
			fmt.Printf("\033[31mError fetching models:\033[0m %v\n", err)
			break
		}

		// Filter for models that support content generation
		supportsGen := false
		for _, method := range model.SupportedGenerationMethods {
			if method == "generateContent" {
				supportsGen = true
				break
			}
		}

		if supportsGen {
			// Remove the "models/" prefix for a cleaner ID
			shortID := strings.TrimPrefix(model.Name, "models/")
			fmt.Printf("\033[32m%-35s\033[0m %s\n", shortID, model.DisplayName)
		}
	}

	fmt.Println("\n\033[34mUsage:\033[0m Update ModelName in gemini.go with one of the IDs above.")
}
